appinsights/src/telemetry/remote_dependency.rs

```diff
@@ -150,7 +181,7 @@ impl From<(TelemetryContext, RemoteDependencyTelemetry)> for Envelope {
             tags: Some(ContextTags::combine(context.tags, telemetry.tags).into()),
             data: Some(Base::Data(Data::RemoteDependencyData(RemoteDependencyData {
                 name: telemetry.name,
-                id: telemetry.id.map(|id| id.to_hyphenated().to_string()),
```

Comment:
Why don't you make this the same way as for RequestTelemetry?

Reply:
The ID is optional in remote dependencies. If the user didn't specify an ID, it's probably fine to leave it empty. An auto-generated ID can't be used to bind the remote dependency to another telemetry.

I'm happy to change it, if you would prefer to generate an ID, though.